core/domain/rule_domain_test.py

```diff
@@ -99,10 +97,48 @@ def test_that_all_rules_have_object_editor_templates(self):
                         param_obj_type.has_editor_js_template(),
                         msg='(%s)' % clazz.description)
                     at_least_one_rule_found = True
-                clses.append(clazz)
 
         self.assertTrue(at_least_one_rule_found)
 
+    def test_that_all_rule_input_fields_have_default_values(self):
+        rule_dir = os.path.join(os.getcwd(), feconf.RULES_DIR)
+
+        at_least_one_rule_found = False
+
+        for loader, name, _ in pkgutil.iter_modules(path=[rule_dir]):
+            if name.endswith('_test') or name == 'base':
+                continue
+            module = loader.find_module(name).load_module(name)
+            for name, clazz in inspect.getmembers(module, inspect.isclass):
+                param_list = rule_domain.get_param_list(clazz.description)
+
+                for (_, param_obj_type) in param_list:
+                    self.assertIsNotNone(
+                        param_obj_type.default_value, msg=(
+                            'No default value specified for object class %s.' %
+                            param_obj_type.__name__))
+                    at_least_one_rule_found = True
+
+        self.assertTrue(at_least_one_rule_found)
+
+    def test_get_default_object_values_function(self):
```

Comment:
@seanlip,
There is no harm from an additional test, just for curiosity:
We want to test that the 'hardcoded' get_default_object_values() is EXACTLY same as extracting default objects from all rules, is it a correct assumption ?

1. test_that_all_rule_input_fields_have_default_values
2. test_get_default_object_values_function

I think, for the current version of tests, if (1) fails (2) will fail, if (2) passes (1) also will pass. Am I correct? (all possible outcomes: {fail, fail}->fail, {fail, pass}->pass, {pass, fail}->fail, {pass, pass}->pass)
So, I think, only (2) is enough to test correctness. 

Sorry, for my curiosity.